assets/prometheus-k8s/rules.yaml

```diff
@@ -300,13 +315,60 @@ spec:
+        or (cluster:master_nodes * on(node) group_left(label_node_hyperthread_enabled)
+        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node)) or on(node) (cluster:hyperthread_enabled_nodes
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node)) or on(node) ((kube_node_labels unless on(node)
+        cluster:hyperthread_enabled_nodes) * on(node) group_left sum(label_replace(node_cpu_info,
+        "node", "$1", "instance", "(.*)")) by (package, node))) BY (label_beta_kubernetes_io_instance_type,
+        label_node_hyperthread_enabled, label_node_role_kubernetes_io)
+      record: cluster:capacity_cpu_sockets_hyperthread_enabled:sum
+    - expr: sum(((cluster:master_infra_nodes * on(node) group_left(label_node_hyperthread_enabled)
+        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
+        or (cluster:master_infra_nodes * on(node) group_left(label_node_hyperthread_enabled)
+        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node)) or on(node) ((cluster:master_nodes * on(node)
+        group_left(label_node_hyperthread_enabled) cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
+        or (cluster:master_nodes * on(node) group_left(label_node_hyperthread_enabled)
+        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node)) or on(node) ((cluster:infra_nodes * on(node)
+        group_left(label_node_hyperthread_enabled) cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled!="true"})
+        or (cluster:infra_nodes * on(node) group_left(label_node_hyperthread_enabled)
+        cluster:hyperthread_enabled_nodes{label_node_hyperthread_enabled="true"})
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node)) or on(node) (cluster:hyperthread_enabled_nodes
+        * on(node) group_left sum(label_replace(node_cpu_info, "node", "$1", "instance",
+        "(.*)")) by (package, node))  or on(node) ((kube_node_labels unless on(node)
+        cluster:hyperthread_enabled_nodes) * on(node) group_left sum(label_replace(node_cpu_info,
+        "node", "$1", "instance", "(.*)")) by (package, node))) BY (label_node_openshift_io_os_id,
+        label_kubernetes_io_arch, label_node_hyperthread_enabled, label_node_role_kubernetes_io_master_infra,
+        label_node_role_kubernetes_io_master, label_node_role_kubernetes_io_infra)
+      record: node_role_os_version_machine:cpu_capacity_sockets:sum
```

Comment:
I get no value when trying this on an existing cluster. It was a 4.1 cluster, is this specific to an openshift version?

Reply:
@LiliC it is dependent on the work @pgier did for 4.3 to pull in `node_exporter` features that capture sockets and hyperthreading enablement:
https://bugzilla.redhat.com/show_bug.cgi?id=1760393#c1
https://bugzilla.redhat.com/show_bug.cgi?id=1760393#c2

These will need to be back ported to support earlier releases.

Reply:
What I see in the links in the description looks good to me.